### ISSUE TITLE
LSF Default Setting from Origen

### ISSUE THREAD

**Issue by coreyeng** [Origen-SDK/origen#284]
Hi,

After seeing @priyavadan's PR for #283, which is quick fix to get IT off his back for not using enough cores per job, I wonder if we shouldn't generalize this even more. The LSF is very site specific, and some LSFs (I say some because I'm not sure how everyone's LSF if implemented) can accepted environment variable parameters as well. For example, when I source a design environment within our company, I get an LSF project and queue to use, allocated to the project. But, origen will always override it with its defaults then pass them in as command line arguments.

I don't think many (if any) defaults should be in LSF, and if it isn't set either by a site config (would need to be added to support this), by the application, or a plugin, then it will fall back to just not adding a switch, which should use any defaults from the environment, or if there's company defaults directly place in LSF. I think Origen might be overstepping its LSF control a bit. Thoughts?

Thanks.

**Comment by priyavadan** [comment]
@coreyeng I believe Origen will overwrite the defaults with the one you have setup in your application.

if it is set to default, currently in NxDI, the submitted jobs will be lost on the LSF as the queue "short" does not exist.

**Comment by coreyeng** [comment]
Rewording a bit:

~~~
 But, origen will always override it with its defaults then pass them in as command line arguments.
~~~

What I meant is that origen will always provide some sort of default, either by itself, a plugin, or the app, but it will always add <i>something</i>, so an environment variable default or a company-internal default will never get used unless somewhere things are explicitly set to <code>nil</code>.

I did check the code again though, and it will ignore options if the parameters are <code>nil</code>. So, that part is fine. I would just like to see a 'default' origen LSF just assume the environment/company settings. Right now, I think you'd need to pass in a bunch of <code>nils</code> to LSF to get it to ignore all options. My reasoning is there's no LSF that will work across all companies, unless they setup their LSF with Origen's default in mind. 

So, I think Origen shouldn't even try. Just let them alone and either let a default plugin or site config handle it. At least with a site config, I can always add a line to the design source setup to account for Origen's (what would probably be something like ORIGEN_LSF_PROJECT) to match designs. I guess that's really no different than setting everything to <code>nil</code> in the app, so maybe I'm blowing this out of proportion... but nonetheless, here's a discussion!

**Comment by ginty** [comment]
The LSF configuration should absolutely be transitioned over to be site config variables.
This is just old code which existed way before we had the concept of site config, but that's definitely where things like this should be done now.
That allows site/company default settings, then the ability to override at the project level (say a directory containing multiple Origen apps), within an application, even right down to individual user environment variables.

**Comment by priyavadan** [comment]
@ginty definitely makes sense. I can append stuff to #283 and add it to the site config variables. 

the project string 'origen' does not exist within NXP, I think we should create one. @coreyeng and I were discussing this yesterday evening as well.

**Comment by coreyeng** [comment]
Fixed in #283 Thanks!
